paasta_tools/vitesscluster_tools.py

```diff
@@ -15,6 +17,594 @@
+        "value": "0",
+    },
+    {
+        "name": "DB",
+        "value": "",
+    },
+    {
+        "name": "EXTERNAL_DB",
+        "value": "1",
+    },
+    {
+        "name": "KEYSPACE",
+        "value": "",
+    },
+    {
+        "name": "ROLE",
+        "value": "rdonly",
+    },
+    {
+        "name": "WEB_PORT",
+        "value": WEB_PORT,
+    },
+    {
+        "name": "GRPC_PORT",
+        "value": GRPC_PORT,
+    },
+    {
+        "name": "TOPOLOGY_FLAGS",
+        "value": "",
+    },
+    {
+        "name": "VAULT_ADDR",
+        "value": "https://vault-dre.uswest1-devc.yelpcorp.com:8200",
+    },
+    {
+        "name": "VAULT_ROLEID",
+        "valueFrom": {
+            "secretKeyRef": {
+                "name": "paasta-vitessclusters-secret-vitess-k8s-vault-vttablet-approle-roleid",
+                "key": "vault-vttablet-approle-roleid",
+            }
+        },
+    },
+    {
+        "name": "VAULT_SECRETID",
+        "valueFrom": {
+            "secretKeyRef": {
+                "name": "paasta-vitessclusters-secret-vitess-k8s-vault-vttablet-approle-secretid",
+                "key": "vault-vttablet-approle-secretid",
+            }
+        },
+    },
+    {
+        "name": "VAULT_CACERT",
+        "value": "/etc/vault/all_cas/acm-privateca-uswest1-devc.crt",
+    },
+]
+
+# Vault auth related variables
+VTGATE_EXTRA_ENV = [
+    {
+        "name": "VAULT_ADDR",
+        "value": "https://vault-dre.uswest1-devc.yelpcorp.com:8200",
+    },
+    {
+        "name": "VAULT_ROLEID",
+        "valueFrom": {
+            "secretKeyRef": {
+                "name": "paasta-vitessclusters-secret-vitess-k8s-vault-vtgate-approle-roleid",
+                "key": "vault-vtgate-approle-roleid",
+            }
+        },
+    },
+    {
+        "name": "VAULT_SECRETID",
+        "valueFrom": {
+            "secretKeyRef": {
+                "name": "paasta-vitessclusters-secret-vitess-k8s-vault-vtgate-approle-secretid",
+                "key": "vault-vtgate-approle-secretid",
+            }
+        },
+    },
+    {
+        "name": "VAULT_CACERT",
+        "value": "/etc/vault/all_cas/acm-privateca-uswest1-devc.crt",
+    },
+]
+
+
+# Extra Flags
+VTADMIN_EXTRA_FLAGS = {"grpc-allow-reflection": "true"}
+
+VTCTLD_EXTRA_FLAGS = {
+    "disable_active_reparents": "true",
+    "security_policy": "read-only",
+}
+
+VTTABLET_EXTRA_FLAGS = {
+    "log_err_stacks": "true",
+    "grpc_max_message_size": "134217728",
+    "init_tablet_type": "replica",
+    "queryserver-config-schema-reload-time": "1800",
+    "dba_pool_size": "4",
+    "vreplication_heartbeat_update_interval": "60",
+    "vreplication_tablet_type": "REPLICA",
+    "keep_logs": "72h",
+    "enable-lag-throttler": "true",
+    "throttle_check_as_check_self": "true",
+    "throttle_metrics_query": "",
+    "throttle_metrics_threshold": "",
+    "db_charset": "utf8mb4",
+    "disable_active_reparents": "true",
+}
+
+
+def build_affinity_spec(paasta_pool):
```

Comment:
and small naming note in case we're interested in being consistent with the paasta service code (and the other operators) - these types of functions are usually named get_X in the rest of paasta (e.g., get_node_affinity, get_extra_labels, etc)

Reply:
> that said: for paasta services we tend to use the k8s client models (and then convert those to dicts if necessary) - that would probably be more type-safe (and safer in general) than raw dicts

ack. Will definitely add typed arguments to functions. Makes sense to use client models than raw dicts which we did in templates.py, which could be the majority of the rewrite but useful. ++

Reply:
https://github.com/Yelp/paasta/pull/3889/commits/c7a6b7f7665b11eb4e812cba4a46457b851040cd